views/js/controller/list/index.js

```diff
@@ -84,66 +95,83 @@ define(['jquery', 'i18n', 'helpers', 'ui/feedback', 'layout/section', 'css!taoBa
-                                    }else{
-                                        feedback().error(__('Element not deleted'));
-                                    }
+                    const $element = $(this).parent();
+                    const $input   = $element.find('input:text');
+                    const eltUri   = $input.attr('name').replace(/^list-element_([0-9]*)_/, '');
+
+                    const deleteLocalElement = () => {
+                        $element.remove();
+                        feedback().success(__('Element deleted'));
+                    };
+
+                    const deleteServerAndLocalElement = () => {
+                        $.postJson(
+                            delEltUrl,
+                            { uri : eltUri },
+                            response => {
+                                if (response.deleted) {
+                                    deleteLocalElement();
+                                } else {
+                                    feedback().error(__('Element not deleted'));
                                 }
-                            );
+                            }
+                        );
+                    };
+
+                    const deleteElement = () => {
+                        if (eltUri) {
+                            deleteServerAndLocalElement();
                         } else {
-                            $element.remove();
-                            feedback().success(__('Element deleted'));
+                            deleteLocalElement();
                         }
+                    };
+
+                    if ($input.val() === '') {
+                        deleteElement();
+                    } else {
+                        dialogConfirm(
+                            __('Please confirm you want to delete this list element.'),
+                            deleteElement
+                        );
                     }
                 });
             });
 
             $('.list-delete-btn').click(function () {
-                if (confirm(__("Please confirm you want to delete this list. This operation cannot be undone."))) {
-                    var $btn = $(this),
-                        uri = $btn.data('uri'),
-                        $list = $(this).parents(".data-container");
-                    $.postJson(
-                        delListUrl,
-                        {uri: uri},
-                        function (response) {
-                            if (response.deleted) {
-                                feedback().success(__('List deleted'));
-                                $list.remove();
-                            }else{
-                                feedback().error(__('List not deleted'));
+                const $btn  = $(this);
+                dialogConfirm(
+                    __('Please confirm you want to delete this list. This operation cannot be undone.'),
+                    function accept() {
```

Comment:
can be const accept = () => { ...

Reply:
It can, but that would be an ESLint error (no-undef) and my version isn't :)
I wanted to keep the `accept` function name because it self-documents what the code does.